pandas/tests/frame/test_api.py

```diff
@@ -76,21 +76,25 @@ def test_get_value(self):
                 tm.assert_almost_equal(result, expected)
 
     def test_add_prefix_suffix(self):
-        with_prefix = self.frame.add_prefix('foo#')
-        expected = pd.Index(['foo#%s' % c for c in self.frame.columns])
-        tm.assert_index_equal(with_prefix.columns, expected)
-
-        with_suffix = self.frame.add_suffix('#foo')
-        expected = pd.Index(['%s#foo' % c for c in self.frame.columns])
-        tm.assert_index_equal(with_suffix.columns, expected)
-
-        with_pct_prefix = self.frame.add_prefix('%')
-        expected = pd.Index(['%{}'.format(c) for c in self.frame.columns])
-        tm.assert_index_equal(with_pct_prefix.columns, expected)
-
-        with_pct_suffix = self.frame.add_suffix('%')
-        expected = pd.Index(['{}%'.format(c) for c in self.frame.columns])
-        tm.assert_index_equal(with_pct_suffix.columns, expected)
+        with pytest.warns(FutureWarning):
```

Comment:
@jreback @jorisvandenbossche : Are we still going with `tm.assert_produces_warning`, or is using the `pytest` construct okay to use?

Reply:
yes I prefer to use our ``tm.assert_produces_warning`` for now (if we change then we should simply globally change, but in a separate set of PRs)